src/axom/core/detail/FlatTable.hpp

```diff
@@ -120,6 +120,24 @@ struct GroupBucket
     return InvalidSlot;
   }
 
+  template <typename Func>
+  AXOM_HOST_DEVICE int visitHashOrEmptyBucket(std::uint8_t hash, Func&& visitor) const
```

Comment:
Please document this function

In particular, what sort of functor is this function expecting? It seems like it calls `visitor` on all buckets until it finds an empty one, and then eventually either returns the id of an empty bucket or `InvalidSlot`